src/comp/comp.py

```diff
@@ -21,51 +21,55 @@ def __repr__(self):
     Human("David", 31),
 ]
 
+print(humans[0].name)
+print(type(humans[0].age))
+
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with 'D':
 print("Starts with D:")
-a = []
+
+a = [name.name for name in humans if name.name[0] == "D"]
 print(a)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 print("Ends with e:")
-b = []
+b = [name.name for name in humans if name.name[-1] == "e"]
 print(b)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 print("Starts between C and G, inclusive:")
-c = []
+c = [name.name for name in humans if name.name[0] == "C" or name.name[0] == "D" or name.name[0] == "E" or name.name[0] == "F" or name.name[0] == "G"]
 print(c)
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
-d = []
+d = [age.age + 10 for age in humans]
 print(d)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
-e = []
+e = [f"{human.name}-{human.age}" for human in humans]
 print(e)
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
-f = []
+f = [(human.name, human.age) for human in humans if human.age == 27 or human.age == 28 or human.age == 29 or human.age == 30 or human.age == 31 or human.age == 32 ]
```

Comment:
Again, a perfectly valid solution, and a similar way available for you to simplify:

```
if human.age == 27 or human.age == 28 or human.age == 29 or human.age == 30 or human.age == 31 or human.age == 32
```

can be reduced to

```
if human.age in range(27, 33)
```

and therefore

```
f = [(human.name, human.age) for human in humans if human.age in range(27, 33)]
```

Really nice job with this again 🎉